oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUpdate.java

```diff
@@ -274,16 +284,35 @@ private static boolean hasAnyHiddenNodes(NodeBuilder builder){
         return false;
     }
 
+    private void sparseLoggger(String primaryType, String name) {
+        synchronized (countLock) {
```

Comment:
I don't really see a point in synchronizing it... it's just a warning message, why is it important that it's synchronized? Worst case (very unlikey), we see two warning messages, right? Or is there some deeper reason?

Reply:
Well yes, it is not that important to log... will remove synchronize.
There can be double/ no-logs dependent on thread state

Reply:
+1 on removing sync